pages/docs/reference/scope-functions.md

```diff
@@ -0,0 +1,576 @@
+    }
+}
+```
+</div>
+
+#### this
+
+`run`, `with`, and `apply` refer to the context object as a lambda receiver - by keyword `this`. Hence, in their lambdas, the object is available as it would be in ordinary class functions. In most cases, you can omit `this` when accessing the members of the receiver object, making the code shorter. On the other hand, if `this` is omitted, it can be hard to distinguish between the receiver members and external objects or functions. So, having the context object as a receiver (`this`) is recommended for lambdas that mainly operate on the object members: call its functions or assign properties.
+
+<div class="sample" markdown="1" theme="idea">
+```kotlin
+data class Person(var name: String, var age: Int = 0, var city: String = "")
+
+fun main() {
+//sampleStart
+    val adam = Person("Adam").apply { 
+        age = 20                       // same as this.age = 20 or adam.age = 20
+        city = "London"
+    }
+//sampleEnd
+}
+```
+</div>
+
+#### it
+
+In turn, `let` and `also` have the context object as a lambda argument. If the argument name is not specified, the object is accessed by the implicit default name `it`. `it` is shorter than `this` and expressions with `it` are usually easier for reading. However, when calling the object functions or properties you don't have the object available implicitly like `this`. Hence, having the context object as `it` is better when the object is mostly used as an argument in function calls. `it` is also better if you use multiple variables in the code block.
+
+<div class="sample" markdown="1" theme="idea">
+```kotlin
+import kotlin.random.Random
+
+fun writeToLog(message: String) {
+    println("INFO: $message")
+}
+
+fun main() {
+//sampleStart
+    fun getRandomInt(): Int {
+        return Random.nextInt(100).also {
+            writeToLog("getRandomInt() generated value $it")
+        }
+    }
+    
+    val i = getRandomInt()
+//sampleEnd
+}
+```
+</div>
+
+Additionally, when you pass the context object as an argument, you can provide a custom name for the context object inside the scope.
+
+<div class="sample" markdown="1" theme="idea">
+```kotlin
+import kotlin.random.Random
+
+fun writeToLog(message: String) {
+    println("INFO: $message")
+}
+
+fun main() {
+//sampleStart
+    fun getRandomInt(): Int {
+        return Random.nextInt(100).also { value ->
+            writeToLog("getRandomInt() generated value $value")
+        }
+    }
+    
+    val i = getRandomInt()
+//sampleEnd
+}
+```
+</div>
+
+### Return value
+
+The scope functions differ by the result they return:
+* `apply` and `also` return the context object.
+* `let`, `run`, and `with` return the lambda result.
+
+These two options let you choose the proper function depending on what you do next in your code.
+
+#### Context object 
+
+The return value of `apply` and `also` is the context object itself. Hence, they can be used as steps in chains of function calls on the object.  
```

Comment:
It is not clear, what's different from `let` and `run` that can also be used in the middle of call chains. Perhaps, we can call them "side steps" (not sure, maybe "tee steps", or "branch steps"), because they return the same context object on which we can continue chaining calls.

Reply:
Updated the description